glom/core.py

```diff
@@ -636,10 +636,18 @@ def from_text(cls, text):
         Path('a', 'b', 'c')
 
         """
+        def create():
+            segs = text.split('.')
+            segs = [
+                _T_STAR if seg == '*' else
+                _T_STARSTAR if seg == '**' else seg
+                for seg in segs]
```

Comment:
change this:

1- default, one time warning (don't hit warnings module every time because perf)

2- global flag (PATH_STARS) -- default False, if True you get new behavior instead of the warning